File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
         int left = 0, right = s.length();

         while(left <= right){
            if(s[left] < 48 || (s[left] > 57 && s[left] < 65) || (s[left] > 90 && s[left] < 97) || s[left] > 122){
                left ++;
                continue;
            }
            if(s[right] < 48 || (s[right] > 57 && s[right] < 65) || (s[right] > 90 && s[right] < 97) || s[right] > 122){
                right --;
                continue;
            }

            if(tolower(s[left]) != tolower(s[right])){
                return false;
            }

            left++;
            right--;
         }

         return true;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        result = []
        for i in range(len(nums) - 2):
            if i > 0 and nums[i] == nums[i-1]:
                continue

            target = -nums[i]
            left = i + 1
            right = len(nums) - 1
            while left < right:
                sum3 = nums[left] + nums[right]
                if sum3 < target:
                    left += 1
                    if left == i:
                        left += 1
                elif sum3 > target:
                    right -= 1
                    if right == i:
                        right -= 1
                else:
                    result.append([nums[left], nums[right], nums[i]])
                    while left < right and nums[left] == nums[left + 1]:
                        left += 1
                    while left < right and nums[right] == nums[right - 1]:
                        right -= 1
                    left += 1
                    right -= 1
        return result